core/admin.py

```diff
@@ -45,6 +51,9 @@ def enroll(self, blackboard_file, offering):
     def save_model(self, request, obj, form, change):
         super().save_model(request, obj, form, change)
         self.enroll(form.cleaned_data.get('blackboard_file'), obj)
+        if obj.repo_status == Offering.RepoStatus.READY and form.cleaned_data.get('update_repository'):
```

Comment:
This part is a little confusing to me. The `#!python update_repository(obj)` call is inside an `#!python if` that checks the repo status. The `#!python clone_repository(obj)` isn't. So it isn't clear why it tries to update only under some circumstances, but always tries to clone.

Reply:
What I'm suggesting is: either move all checks to the repository functions, or bring them all here.